subprojects/core/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationRoles.java

```diff
@@ -69,7 +69,14 @@ public enum ConfigurationRoles implements ConfigurationRole {
      *
      * AKA {@code INTENDED_DECLARABLE}.
      */
-    INTENDED_BUCKET(false, false, true);
+    INTENDED_BUCKET(false, false, true),
+
+    /**
+     * Meant for configurations which cannot do anything.
+     *
+     * Used for configurations which will be removed in future versions.
+     */
+    NONE(false, false, false);
```

Comment:
❗ I really don't think we should add this mechanism until there is a need to use it.  It seems unlikely we'll want to completely remove a configuration.  Do you have any examples of where you'd want to use this?

Reply:
Yes. We will be completely removing the `archives` configuration, and hopefully the `default` configuration in 9.0

Reply:
I am open to alternatives, but using `NONE` + `ConfigurationRolesForMigration` seemed to be a natural way to get the "fully deprecated" role we need.

Reply:
But it's the wrong mechanism.  You're not migrating to a _role_ that has no valid usage, you're removing the `archives` configuration entirely.  Deprecating its allowed usages doesn't indicate that.  The messages a build author sees won't clue them into the root concern.

Also, won't this spam the logs with lots of warnings whenever you or a plugin your using do something interesting with `confgurations.all{}`?

I think instead you'd want to add some special case warnings on the most important methods that are called on it.  For instance, in `resolve` and `add(Dependency)` and the `attributes` methods first check if name == `archives` and if so warn the configuration is to be removed.  And track emitting these warnings to make sure we only do it once per configuration to be removed per build.

Either way, I'd like to see it in a separate PR, along with testing it on `archives` there, to see what happens.

Reply:
I can remove the NONE part for now until we find a better way to do this. Though we will need a way to do it. 

The `configurations.all` problem still comes about for any configuration which is deprecated for anything. Say we have a `RESOLVABLE_BUCKET_TO_RESOLVABLE`, a `configurations.all` which checks for `bucket` configurations would spam the logs too. 

As for only adding warnings on special cases, I don't think that's enough. We need to warn if the user calls any method on the configuration at all. Ideally, a configuration which is deprecated for all three usages would satisfy that use-case

Reply:
Removed NONE